### ISSUE TITLE
AutoComplete: Enable ctrl+click/shift+click when using multiple='true'

### ISSUE THREAD

**Issue by afunduk** [primefaces/primefaces#11570]
### Description

When using the autocomplete component in multiple mode, users should be able to select multiple similar suggestions at same time. Currently you have to type the same filter characters several times if you want to select more than one suggestion that match the filter characters.

### Describe the solution you would like

When suggestions are shown, users should be able to use control+click or shift+click to select multiple suggestions at the same time without having to retype the required number of filter characters. Some components already have this (e.g. SelectManyMenu and PickList).

**Comment by melloware** [comment]
Agreed this does seem like a good UX addition.

**Comment by melloware** [comment]
@afunduk here is a MonkeyPatch to enable CTRL+CLICK on multiple now.

I did not implement SHIFT+CLICK as it was difficult and confusing since you really don't have anything selected to SHIFT+CLICK between.

```js
if (PrimeFaces.widget.AutoComplete) {
    PrimeFaces.widget.AutoComplete.prototype.bindDynamicEvents = function() {
        var $this = this;

        //visuals and click handler for items
        this.items.off('click.autocomplete mousedown.autocomplete mouseover.autocomplete')
            .on('mouseover.autocomplete', function() {
                var item = $(this);

                if (!item.hasClass('ui-state-highlight')) {
                    $this.items.filter('.ui-state-highlight').removeClass('ui-state-highlight').attr('aria-selected', false);
                    $this.highlightItem(item, true);

                    if ($this.cfg.itemtip) {
                        $this.showItemtip(item);
                    }
                }
            })
            .on('click.autocomplete', function(event) {
                var item = $(this),
                    isMoreText = item.hasClass('ui-autocomplete-moretext');

                if (isMoreText) {
                    $this.input.trigger('focus');
                    $this.invokeMoreTextBehavior();
                } else {
                    $this.addItem(item);
                }

                // #11570 allow CTRL+Click to select multiple
                if (!$this.cfg.multiple || !PrimeFaces.utils.isMetaKey(event)) {
                    $this.hide();
                }
            })
            .on('mousedown.autocomplete', function() {
                $this.preventInputChangeEvent = true;
                $this.checkMatchedItem = false;
            });

        if (this.cfg.showEmptyMessage) {
            this.panel.on('click.emptyMessage', function() {
                if (!this.children) {
                    return;
                }
                var item = $(this.children[0]),
                    isEmptyMessage = item.hasClass('ui-autocomplete-empty-message');

                if (isEmptyMessage) {
                    $this.invokeEmptyMessageBehavior();
                }
            });
        }

        if (PrimeFaces.env.browser.mobile) {
            this.items.on('touchstart.autocomplete', function() {
                if (!$this.touchToDropdownButton) {
                    $this.itemClick = true;
                }
            });
        }
    }
}
```
